### PR TITLE
LPS-150314 Encapsulate logic to getSegmentsExperienceId from request on a new util class 

### PR DESCRIPTION
**Pull request by ealonso**
Motivation
=======
Since we are going to convert the default experience in a real experience as part of https://issues.liferay.com/browse/LPS-147302, we are trying to extract as much code as possible to common places, to make this task easier.

In this case the idea is to have a common place with the related logic for getting the segments experience id from the request. 

[Link to the issue](https://issues.liferay.com/browse/LPS-150314) 

Proposed Solution
========
Creates a new util class **SegmentsExperienceUtil** and moves the common logic to get the segments experience id from the request, this information is in this request attribute **SegmentsWebKeys.SEGMENTS_EXPERIENCE_IDS**. 

Updates the usages of **SegmentsWebKeys.SEGMENTS_EXPERIENCE_IDS** with the new method.

Includes tests to validate the correct behavior of the new class and method.

Thanks!!

### CODE REVIEW

**Comment by ealonso** [comment]
ci:tes:sf

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:test:sf

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:test:sf

**Comment by ealonso** [comment]
ci:test:relevant

File: modules/apps/segments/segments-api/src/main/java/com/liferay/segments/util/SegmentsExperienceUtil.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		if (ArrayUtil.isNotEmpty(segmentsExperienceIds)) {
+			return segmentsExperienceIds[0];
+		}
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		return SegmentsExperienceLocalServiceUtil.
+			fetchDefaultSegmentsExperienceId(themeDisplay.getPlid());
```
Comment (cgoncas) [inline modules/apps/segments/segments-api/src/main/java/com/liferay/segments/util/SegmentsExperienceUtil.java:47]: We usually follow 2 approaches:

1. Create a Manager class, that is instantiated using the SegmentsExperienceLocalService.

2. Make getSegmentsExperienceId receive the SegmentsExperienceLocalService.

We usually prefer the first option, so we can test it easily, and all the dependencies are explicit.

Thanks!!!

Comment (ealonso) [reply modules/apps/segments/segments-api/src/main/java/com/liferay/segments/util/SegmentsExperienceUtil.java:47]: Hey @cgoncas,

Do you have an example of the Manager class?

Thanks!!

Comment (cgoncas) [reply modules/apps/segments/segments-api/src/main/java/com/liferay/segments/util/SegmentsExperienceUtil.java:47]: For example the [AnalyticsReportsDataProvider](https://github.com/liferay/liferay-portal/blob/f4753776823a66409290376241d024ef5f6d44c0/modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProvider.java), follow this pattern.

**Comment by ealonso** [comment]
Hey @cgoncas,

I've updated the pr with your recommendations and I've created a new manager class for SegmentsExperience.

Please let me know if you need something else.

Thanks!!

**Comment by ealonso** [comment]
ci:test:sf

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
> Hey @cgoncas,
> 
> I've updated the pr with your recommendations and I've created a new manager class for SegmentsExperience.
> 
> Please let me know if you need something else.
> 
> Thanks!!

Awesome @ealonso !!

Everything looks good :)

**Review by cgoncas — APPROVED** [review]
LGTM, lets forward it once the tests results are in green.

**Comment by cgoncas** [comment]
ci:test:tango

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:test:sf

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Pull request closed without merging**